lib/src/revset_graph.rs

```diff
@@ -93,3 +105,180 @@ impl Iterator for ReverseRevsetGraphIterator {
         self.items.pop()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::cmp::Ordering;
+
+    use itertools::Itertools as _;
+
+    use super::*;
+    use crate::backend::ObjectId;
+
+    fn id(c: char) -> CommitId {
+        let d = u8::try_from(c).unwrap();
+        CommitId::new(vec![d])
+    }
+
+    fn missing(c: char) -> RevsetGraphEdge {
+        RevsetGraphEdge::missing(id(c))
+    }
+
+    fn direct(c: char) -> RevsetGraphEdge {
+        RevsetGraphEdge::direct(id(c))
+    }
+
+    fn indirect(c: char) -> RevsetGraphEdge {
+        RevsetGraphEdge::indirect(id(c))
+    }
+
+    fn format_edge(edge: &RevsetGraphEdge) -> String {
+        let c = char::from(edge.target.as_bytes()[0]);
+        match edge.edge_type {
+            RevsetGraphEdgeType::Missing => format!("missing({c})"),
+            RevsetGraphEdgeType::Direct => format!("direct({c})"),
+            RevsetGraphEdgeType::Indirect => format!("indirect({c})"),
+        }
+    }
+
+    fn format_graph(
```

Comment:
> Or maybe we can port AsciiGraphDrawer to the lib crate instead?

I would like to delete AsciiGraphDrawer since we now have Sapling's `renderdag`. I don't remember when we changed the default, but it feels like it's been long enough that we can do that now. Regardless of when we do that, would it make sense to use `renderdag` in this PR?

Reply:
Switched to `renderdag`. The test output gets more readable.